src/apt-map.ts

```diff
@@ -1,218 +1,297 @@
   }
 
-  get isEmpty(): boolean {
-    return this.size === 0;
+  /**
+   * Returns `true` if apt map does not have any elements, otherwise `false`.
+   */
+  get isEmpty() {
+    return this.size === 0
   }
 
-  get lastKey(): K | undefined {
-    const index = this.size - 1;
-    const keys = this.keysAsArray();
+  /**
+   * Returns last key if it exists, otherwise `undefined`.
+   */
+  get lastKey() {
+    const index = this.size - 1
+    const keys = this.keysAsArray()
 
     return keys[index]
   }
 
-  get lastValue(): V | undefined {
-    const index = this.size - 1;
-    const values = this.valuesAsArray();
+  /**
+   * Returns last value if it exists, otherwise `undefined`.
+   */
+  get lastValue() {
+    const index = this.size - 1
+    const values = this.valuesAsArray()
 
     return values[index]
   }
 
-  clone(): AptMap<K, V> {
-    return new AptMap<K, V>(this.entriesAsArray());
+  /**
+   * Returns `true` if a condition in callback function is satisfied for all entries, otherwise `false`.
+   */
+  every(cb: Callback<K, V>) {
+    let result = false
+
+    for (const [mapKey, mapValue] of this) {
+      result = cb(mapValue, mapKey, this)
+
+      if (!result)
+        break
+    }
+
+    return result
   }
 
-  concat<T extends V>(aptMap: AptMap<K, T>): AptMap<K, V> {
+  /**
+   * Creates a shallow copy of an AptMap and returns it.
+   */
+  clone() {
+    return new AptMap<K, V>(this.entriesAsArray())
+  }
+
+  /**
+   * Concatenates 2 apt maps and return merged result.
+   */
+  concat<T extends V>(aptMap: AptMap<K, T>) {
     return this
       .clone()
       .setMany<T>(aptMap)
   }
 
-  entriesAsArray(): Array<[K, V]> {
-    return Array.from(this.entries());
+  /**
+   * Returns an array of [key, value] pairs.
+   */
+  entriesAsArray() {
+    return Array.from(this.entries())
   }
 
-  every(cb: (value: V, key?: K, aptMap?: AptMap<K, V>) => boolean): boolean {
-    let result = false;
+  /**
+   * Filters input apt map using callback function and returns filtered apt map.
+   */
+  filter(cb: Callback<K, V>) {
+    const filtered = new AptMap<K, V>()
 
-    for (const [mapKey, mapValue] of this) {
-      result = cb.apply(this, [mapValue, mapKey, this]);
+    this.forEach((mapValue, mapKey) => {
+      const isFiltered = cb(mapValue, mapKey, this)
 
-      if (!result) {
-        break;
-      }
-    }
+      if (isFiltered)
+        filtered.set(mapKey, mapValue)
+    })
 
-    return result;
+    return filtered
   }
 
-  filter(cb: (value: V, key?: K, aptMap?: AptMap<K, V>) => boolean): AptMap<K, V> {
-    const filtered = new AptMap<K, V>();
+  /**
+   * Returns first entry which satisfies condition in callback function, otherwise an empty array.
+   */
+  findEntry(cb: Callback<K, V>) {
+    let foundKey: K | undefined
+    let foundValue: V | undefined
 
-    for (const [mapKey, mapValue] of this) {
-      const isFiltered = cb.apply(this, [mapValue, mapKey, this]);
+    this.forEach((mapValue, mapKey) => {
```

Comment:
With regular `for...of` we used `break;` which used to stop the loop when the item is found. Now, with your change it continues loop. So when you have 1000 entries and found entry is first entry you would still interate through other 999. 

That's why `for...of` was applied for all methods (even without `brake;`) just to have all in same style

Reply:
![image](https://github.com/jarcko/apt-map/assets/15153245/50bcccc0-94e5-42aa-a870-3afe39f6efbe)

You can check differences if you want.